Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractCompactRandomCutTree.java

```diff
@@ -150,9 +103,80 @@ protected void addSequenceIndex(Integer nodeRef, long sequenceIndex) {
         sequenceIndexes[leafRef].put(sequenceIndex, num + 1);
     }
 
+    public double getBoundingBoxCacheFraction() {
+        return boundingBoxCacheFraction;
+    }
+
+    abstract void swapCaches(int[] map);
+
+    public void renormalize() {
```

Comment:
Let's stick to either "canonical" or "normalized" throughout.

Should we add a method to NodeStore to "canonicalize" in place? It could return the index map:

```
public int[] canonicalize(PointRef root);
```

Reply:
Making the root a part of the nodestore will help this ...

Reply:
Solved partially; the canonicalAndNotALeaf() checks if the root = 0 in the nodestore